service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/RepresentativesInbox.java

```diff
@@ -47,11 +56,44 @@
                 .collect(Collectors.toCollection(LinkedHashSet::new))
         );
 
-
         return emails;
     }
 
-    private boolean shouldSend(RepresentativeServingPreferences preference, Element<? extends WithSolicitor> element) {
+    public Set<?> getNonSelectedRespondentsRecipients(
+        RepresentativeServingPreferences servingPreferences,
+        CaseData caseData,
+        List<Element<Respondent>> respondentsSelected,
+        Function<Element<Representative>, ?> mapperFunction) {
+
+        Set<UUID> representativeIds = caseData.getAllRespondents().stream()
+            .flatMap(respondentElement -> respondentElement.getValue().getRepresentedBy()
+                .stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        Set<UUID> selectedRepresentativeIds = respondentsSelected.stream()
+            .flatMap(otherElement -> otherElement.getValue().getRepresentedBy().stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        return caseData.getRepresentativesElementsByServedPreference(servingPreferences)
+            .stream()
+            .filter(representativeElement -> representativeIds.contains(representativeElement.getId()))
+            .filter(representativeElement -> !selectedRepresentativeIds.contains(representativeElement.getId()))
+            .map(mapperFunction)
+            .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    public Set<Recipient> getSelectedRecipientsWithNoRepresentation(List<Element<Respondent>> selectedRespondents) {
+        return selectedRespondents.stream()
+            .map(Element::getValue)
+            .filter(respondent -> isEmpty(respondent.getRepresentedBy())
+                && !isNull(respondent.getParty().getAddress())
```

Comment:
could you use something like isNotNull here?

Reply:
which library has isNotNull? I can see only isNotEmpty()

Reply:
this code no longer exists, I've replaced it with `respondent.hasAddress()`